iothub/device/src/Transport/AmqpIot/AmqpIotReceivingLink.cs

```diff
@@ -215,50 +212,12 @@ private void OnTwinChangesReceived(AmqpMessage amqpMessage)
 
                         // Retry for Http status code request timeout, Too many requests and server errors
                         var exception = new IotHubClientException(error, status >= 500 || status == 429 || status == 408);
-                        _onTwinMessageReceived.Invoke(null, correlationId, null, exception);
+                        _onTwinMessageReceived.Invoke(null, correlationId, exception);
                     }
                 }
                 else
                 {
-                    if (correlationId == null)
```

Comment:
Why is this whole block no longer needed?

Reply:
This block was handling the deserialization into different types and sending them over to the transport layer. Now, we pass in the AmqpMessage directly and the transport layer handles these different scenarios.

desired props: https://github.com/Azure/azure-iot-sdk-csharp/blob/087d50093a0dd1458018b53d7bec218cfc81a3c1/iothub/device/src/Transport/Amqp/AmqpTransportHandler.cs#L459-L472
get twin and update reported property response: https://github.com/Azure/azure-iot-sdk-csharp/blob/087d50093a0dd1458018b53d7bec218cfc81a3c1/iothub/device/src/Transport/Amqp/AmqpTransportHandler.cs#L475-L499
desired prop subscription ack: https://github.com/Azure/azure-iot-sdk-csharp/blob/087d50093a0dd1458018b53d7bec218cfc81a3c1/iothub/device/src/Transport/Amqp/AmqpTransportHandler.cs#L500-L505